### PR TITLE
Add experimental typed BQ Taps

### PR DESCRIPTION
**Pull request by ravwojdyla**

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala [hunk 1]
```diff
@@ -233,4 +233,30 @@ package object experimental {
 
   }
 
+
+  /** Enhanced version of [[Taps]] with type-safe features for BigQuery [[Taps]]. */
+  implicit class TypedBigQueryTaps(self: Taps) {
+
+    /** Get a `Future[Tap[T]]` for BigQuery table. */
+    def typedBigQueryTable[T <: HasAnnotation : TypeTag : ClassTag](tableSpec: TableReference)
```
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: Should get table from the provided BQT and rename the argument to new source.

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala [hunk 2]
```diff
@@ -233,4 +233,30 @@ package object experimental {
...
+    : Future[Tap[T]] = {
+      val bqt = BigQueryType[T]
+      import scala.concurrent.ExecutionContext.Implicits.global
+      self.bigQueryTable(tableSpec).map(_.map(bqt.fromTableRow))
+    }
+
+    /** Get a `Future[Tap[T]]` for BigQuery table. */
+    def typedBigQueryTable[T <: HasAnnotation : TypeTag : ClassTag](tableSpec: String)
```
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: Same as above.

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala [hunk 3]
```diff
@@ -233,4 +233,30 @@ package object experimental {
 
   }
 
+
```
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: Or maybe i could even consolidate both table/select to one actually.

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala [hunk 4]
```diff
@@ -233,4 +233,36 @@ package object experimental {
...
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: nitpick, extra blank line?

**Pull request merged**